src/hotspot/share/gc/serial/serialFullGC.cpp

```diff
@@ -365,10 +365,11 @@ class Compacter {
         cur_addr += relocate(cur_addr);
       }
 
+      HeapWord* new_top = get_compaction_top(i);
       // Reset top and unused memory
-      space->set_top(get_compaction_top(i));
-      if (ZapUnusedHeapArea) {
-        space->mangle_unused_area();
+      space->set_top(new_top);
```

Comment:
I'd rather the new_top variable declaration was after the comment rather than before.